core/domain/user_jobs_one_off.py

```diff
@@ -470,3 +471,142 @@ def map(user_settings_model):
     def reduce(key, values):
         """Implements the reduce function for this job."""
         yield (key, len(values))
+
+
+class CleanUpUserSubscribersModelOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that cleans up UserSubscribersModel by removing user id if it is
+    present in subscriber ids.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSubscribersModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        if item.id in item.subscriber_ids:
+            item.subscriber_ids.remove(item.id)
+            item.put()
+            yield ('Removed user from their own subscribers list', item.id)
```

Comment:
What code prevents this from happening in the future; do we have a backend test for it?

Reply:
I checked on oppia.org and there is no option for the user to self-subscribe. Further, the extra check for this would be to ensure in backend that [subscribe method](https://github.com/oppia/oppia/blob/develop/core/domain/subscription_services.py#L96) checks that `user_id` of the subscriber is never equal to the `creator_id`.

Reply:
Confirming that no further fix is required - I have fixed this in this PR itself ([Commit](https://github.com/oppia/oppia/pull/10794/commits/8abe91a91b8c79b75d645d7a96a446799de7ec20#diff-c7f5624531b2131c26b7adefe16f716aR106)).